src/retro_data_structures/formats/pak_wii.py

```diff
@@ -62,4 +74,162 @@ def _emitparse_header(code: construct.CodeGen) -> str:
     _resources_end_assert=construct.Check(
         construct.this.table_of_contents.RSHD == construct.this._resources_end - construct.this._resources_start
     ),
+) 
+##TODO : Once the struct is confirmed, compile it
+
+## Not sure how this is supposed to work, is this for the whole compressed resource
+##or a single block ?
+CompressedPakResource = FocusedSeq(
+    "data",
+    decompressed_size = Rebuild(Int32ub, construct.len_(construct.this.data)),
+    ## Since Corruption only uses LZO1X, I don't think
+    ## we should necessarily check which compression algorithm to use
+    data = IfThenElse(
+        game_check.uses_lzo,
+        LZOCompressedBlock(construct.this.decompressed_size),
+        None
+    )
 )
+
+@dataclasses.dataclass
+class PakFile:
+    asset_id: AssetId
+    asset_type: AssetType
+    should_compress: bool 
+    uncompressed_data: bytes | None    
+    compressed_data: bytes | None      
+    extra: construct.Container | None = None
+
+    ## Considering I'm working on corruption specifically, 
```

Comment:
It's best to leave it without a default. In particular, this format is also used for DKCR.